### PR TITLE
From mistral

### PR DESCRIPTION
**Pull request by wassim-trabelsi**
| <a href="https://ellipsis.dev" target="_blank">Ellipsis</a> | :rocket: This PR description was created by [Ellipsis](https://www.ellipsis.dev) for commit e6ebb154ccc9bbaecc3504bb933edf865e05e033.  | 
|--------|--------|

### Summary:
This PR introduces support for the MistralAI API in the instructor library, including a new function `from_mistral`, updates to the `Provider` enum, new tests, an updated example, and new documentation.

**Key points**:
- Added support for MistralAI API in the instructor library.
- Added a new function `from_mistral` in `instructor/client_mistral.py`.
- Imported `from_mistral` in `instructor/__init__.py` if `mistralai` package is available.
- Updated `Provider` enum in `instructor/utils.py` to include `MISTRAL`.
- Added tests for the new functionality in `tests/test_new_client.py`.
- Updated the example in `examples/mistral/mistral.py`.
- Added a new documentation file `docs/examples/mistral.md`.
- Added the new `md` file to `mkdocs.yml`.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

File: tests/test_new_client.py [hunk 1]
```diff
@@ -338,3 +335,33 @@ class Group(BaseModel):
     assert group.members[1].name == "Paul McCartney"
     assert group.members[2].name == "George Harrison"
     assert group.members[3].name == "Ringo Starr"
+
+def test_client_from_mistral_with_response():
```
Comment (jxnl) [inline tests/test_new_client.py:339]: Can we skip tests for now until I get api keys for everything

File: instructor/client_mistral.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import mistralai.client
+import mistralai.async_client as mistralaiasynccli
+import instructor
+from typing import Type, ParamSpec
```
Comment (jxnl) [inline instructor/client_mistral.py:4]: please merge in main and use the proper __future__ imports to make sure this is compatible with 3.9

File: instructor/client_mistral.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+def from_mistral(
+    client: mistralai.client.MistralClient,
+    mode: instructor.Mode = instructor.Mode.MISTRAL_TOOLS,
+    **kwargs,
+) -> instructor.Instructor: ...
+
+
+@overload
+def from_mistral(
+    client: mistralai.client.MistralClient,
```
Comment (jxnl) [inline instructor/client_mistral.py:24]: async

**Comment by wassim-trabelsi** [comment]
I need a workflow approval to successfully merge in main (probably because i added mistral tests with skip decorator). Can you help me with that ?

File: tests/test_new_client.py [hunk 4]
```diff
@@ -357,9 +357,9 @@ def test_client_from_mistral_with_response():
     assert user.age == 10
 
 
-#@pytest.mark.skip(reason="Skip for now")
+@pytest.mark.skip(reason="Skip for now")
 def test_client_mistral_response():
```
Comment (jxnl) [inline tests/test_new_client.py:361]: can move the import in here, as well!

**Comment by jxnl** [comment]
might make sense to add the mistral ai client to the pyproject toml

**Comment by jxnl** [comment]
this looks good to merge! let me know!

**Comment by wassim-trabelsi** [comment]
> this looks good to merge! let me know!

i'm not sure how. I don't think it depends on me right ?

**Pull request merged**